api/settings/v1/recording_rules.proto

```diff
@@ -0,0 +1,70 @@
+syntax = "proto3";
+
+package settings.v1;
+
+import "types/v1/types.proto";
+
+service RecordingRulesService {
+  rpc ListRecordingRules(ListRecordingRulesRequest) returns (ListRecordingRulesResponse) {}
+  rpc InsertRecordingRule(InsertRecordingRuleRequest) returns (InsertRecordingRuleResponse) {}
+  rpc DeleteRecordingRule(DeleteRecordingRuleRequest) returns (DeleteRecordingRuleResponse) {}
+}
+
+message ListRecordingRulesRequest {}
+
+message ListRecordingRulesResponse {
+  repeated RecordingRule_API rules = 1;
+}
+
+message InsertRecordingRuleRequest {
+  string metric_name = 1;
+  repeated string matchers = 2;
+  repeated string group_by = 3;
+  repeated types.v1.LabelPair external_labels = 4;
+  string prometheus_data_source = 5;
+}
+
+message InsertRecordingRuleResponse {
+  RecordingRule_API rule = 1;
+}
+
+message DeleteRecordingRuleRequest {
+  string id = 1;
+}
+
+message DeleteRecordingRuleResponse {}
+
+message RecordingRule_API {
+  string id = 1;
+  string metric_name = 2;
+  string profile_type = 3;
+  repeated string matchers = 4;
+  repeated string group_by = 5;
+  repeated types.v1.LabelPair external_labels = 6;
+  string prometheus_data_source = 7;
```

Comment:
Same here, I am unsure what data_source will be set to?

Is this a Grafana data source uid? How would we use it?

Grafana Datasources usually would only have read permissions, so we probably can't use this as "where should the generated metrics be ingested to"

Reply:
In UI we want to provide an option to the user what data source they want the recording rule to forward the metric to.

Here's the form in the UI:

<img width="749" alt="Screenshot 2025-02-25 at 5 32 14 PM" src="https://github.com/user-attachments/assets/7a5c528e-2609-48e6-a91f-a72eedfd72bc" />

Right now it doesn't allow users to choose the destination of their recording rule, but I would imagine this is something we want to support in the future. Or should I get rid of this field for now and we can add something in later when we expand support for this feature?

Reply:
Removed in https://github.com/grafana/pyroscope/pull/3946/commits/96541dad320965a96e15785f0f1021a300c8e488.